File: practice_java_program/notpad/SelectionSortAssending.java
//Selection Sort - Find minimum element and place it at correct position.
//Time Complexity: O(n²) (all cases)

public class SelectionSortAssending{
	public static void main(String[] args){
		int[] numbers = {4,3,1,5,6,2,9};

		for(int i=0; i<numbers.length-1; i++){
			int minindex=i;//4
			for(int j=i+1; j<numbers.length; j++){
				if (numbers[j]<numbers[minindex]){
					minindex=j;
				}
			}
			int temp=numbers[i];
			numbers[i]=numbers[minindex];
			numbers[minindex]=temp;
		}
		for(int nums:numbers){
			System.out.print(nums+" ");
		}
	}
}

File: python_oriented_interview/notpad/SelectionSortAssending.py
# Selection Sort - Find minimum element and place it at correct position.
# Time Complexity: O(n²) (all cases)

numbers = [4, 3, 1, 5, 6, 2, 9]

for i in range(len(numbers) - 1):
    minindex = i
    for j in range(i + 1, len(numbers)):
        if numbers[j] < numbers[minindex]:
            minindex = j
            
    temp = numbers[i]
    numbers[i] = numbers[minindex]
    numbers[minindex] = temp

for nums in numbers:
    print(nums, end=" ")
print()
